File: 2020/3/1.py
with open("input", "r") as file:
    text = file.read().splitlines()

limit = len(text[0])
xcount = 0
ans = 0

for line in text[1:]:
    xcount+=3
    current = line[xcount % limit]

    ans += 1 if current == "#" else 0

print(ans)

File: 2020/3/1.rs
use std::fs::read_to_string;

fn main() {
    let input = read_to_string("./input").unwrap();

    let vec_lines: Vec<_> = input.lines().collect();

    let limit = vec_lines[0].len();
    let mut xcount = 0;
    let mut ans = 0;

    for line in vec_lines[1..].iter() {
        xcount += 3;
        let current = line.chars().nth(xcount % limit).unwrap();

        ans += if current == '#' {1} else {0};
    }

    println!("{}", ans);
}